semgrep-core/Parsing/Parse_target.ml

```diff
@@ -298,7 +298,12 @@ let just_parse_with_lang lang file =
   | Lang.R ->
       failwith "No R parser yet; improve the one in tree-sitter"
   | Lang.Yaml ->
-      failwith "No Yaml parser yet, parsed only for semgrep in Parse_rule.ml"
+      let ch = open_in file in
+      let s = really_input_string ch (in_channel_length ch) in
+      close_in ch;
+      { ast = Yaml_to_generic.program {str = s; file};
```

Comment:
would be better to read the file in Yaml_to_generic.program, so that the interface of Yaml_to_generic.program
remains simple (Common.filename -> AST_generic.program)